spyql/processor.py

```diff
@@ -44,38 +45,56 @@ def init_vars():
         spyql.log.user_debug(f"Init file not found: {init_fname}")
     except Exception as e:
         spyql.log.user_warning(f"Could not load {init_fname}", e)
+    finally:
+        # user defined variables
+        vars["user_query_vars"] = user_query_vars
     return vars
 
 
 class Processor:
     @staticmethod
-    def make_processor(prs, strings, input_options):
+    def make_processor(prs, strings, interactive = False, input_options = {}):
         """
         Factory for making a file processor based on the parsed query
         """
+
+        spyql.log.user_info(f"Input Options: {input_options}")
+
+        # Since I did not want to damage the existing flow using FROM, when the query
+        # is *really* interactive (as determined by Q), we set "interactive" in the
+        # parsed dict. This is different from interactive passed in the spec above,
+        # which tells the processors that you have to read data from filepath and not
+        # from stdin.
+        if prs.get("interactive", None):
+            return InteractiveProcessor(prs=prs, strings=strings, **input_options)
```

Comment:
```suggestion
```
I might me overlooking something, but it seems to me that we do not need an `InteractiveProcessor`. The  `PythonExprProcessor` would do just fine since, with the change I suggest in the `init_vars` method, all variables are directly available in the scope of the spyql query.